### PR TITLE
docs(toh-6/ts): minor edits and enhancements

### PR DESCRIPTION
**Pull request by chalin**
Changes to prose:

- Complete TODO item of displaying `heroes.component` errors.
- Mainly copyedits.
- Add of `block` statements so that prose can be used on Dart side.
- Show excerpt and briefly explain of changes (previously missing):
  - `app/hero-detail.component.html`
  - `app/heroes.component.ts` error handling
- Add missing file to changed/added files listing and makeTabs
  - `toh-6/ts/app/in-memory-data.service.ts,
  - `toh-6/ts/sample.css`

Code changes:

- Mainly copyedits
- Renamed `heroes.component.ts` `delete` to `deleteHero` to match
naming of other methods

Contributes to #1628.

### CODE REVIEW

**Comment by chalin** [comment]
@kwalrath @Foxandxss @johnpapa @wardbell : ready for review. As usual, because of the addition of blocks causes changes in indentation, it is best to review changes by ignoring whitespace, like [this](https://github.com/angular/angular.io/pull/1686?w=1). Please see the original PR comment for a description of changes (besides copyedits).

**Comment by kwalrath** [comment]
The copyedits look reasonable to me, but I'll leave the content review to the TS experts.

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 1]
```diff
@@ -62,153 +77,161 @@ code-example(language="bash").
...
-    Skip it when you have a real web api server.
++makeExample('app/in-memory-data.service.ts', 'init')
 
+p This file replaces the #[code #[+adjExPath('mock-heroes.ts')]] which is now safe to delete.
+
+block dont-be-distracted-by-backend-subst
+  .alert.is-helpful
+    :marked
+      This chapter is an introduction to the !{_Angular_http_library}.
+      Please don't be distracted by the details of this backend substitution. Just follow along with the example.
```
Comment (Foxandxss) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: I personally don't like this two lines. Perhaps the tone.

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 2]
```diff
@@ -62,153 +77,161 @@ code-example(language="bash").
...
 
-    Adjust the code to match *your web api*.
+    Adjust the code to match *your web API*.
 :marked
-  The caller is unaware of these machinations. It receives a promise of *heroes* just as it did before.
-  It has no idea that we fetched the heroes from the server.
-  It knows nothing of the twists and turns required to turn the http response into heroes.
+  The caller is unaware of these machinations. It receives a !{_Promise} of *heroes* just as it did before.
+  It has no idea that we fetched the heroes from the (mock) server.
+  It knows nothing of the twists and turns required to convert the HTTP response into heroes.
```
Comment (Foxandxss) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: Personally I don't like either `turn` nor `convert`. I would personally say `map` which is a programming concept too.

**Comment by Foxandxss** [comment]
I left two comments on the prose, but the rest are good to me 👍

File: public/docs/ts/latest/tutorial/toh-pt6.jade (hunk 1 above)
Comment (chalin) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: I agree, but that was in the original TS text so I left it as is (but excluded it from the Dart edition). Remember that for this time, my goal was mainly to proof read, not a general edit.

File: public/docs/ts/latest/tutorial/toh-pt6.jade (hunk 2 above)
Comment (chalin) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: I agree that it is all a bit colloquial ("twists and turns"). "Convert" seems at least a step in the right direction as compared to "turn" :). Maybe we can revise this on the next pass?

**Comment by chalin** [comment]
@kwalrath @Foxandxss : thank you both for reviewing this.
@naomiblack : it would be great if this could be merged before @johnpapa starts carving out the CLI editions.

**Comment by chalin** [comment]
I was reviewing my notes and noticed I had forgotten to remove an unnecessary change to a .css, relative to toh-5 (new `#docregion` that wasn't used). I just pushed a commit with that change.

**Comment by Adam-Meisen** [comment]
Is `sample.css` ever explicitly mentioned in the tutorial? I only realized it existed after I noticed that my delete buttons didn't look like they do in the examples. In fact, styling the delete button is all that file does.

**Comment by chalin** [comment]
@kwalrath : rebased.

Hi @Adam-Meisen : while it is true that `sample.css` is not mentioned in the prose, using a `sample.css` file is a convention followed in other chapters in the Dev Guide and the Cookbook.

It may make sense, as you have proposed, to merge the `sample.css` styles into one of the component's styles, but before doing so it may make sense to determine if there was a reason for doing it the way it is now.

**Comment by Adam-Meisen** [comment]
But then why does the file just appear at the end of part 6 without being
mentioned? Why does a stylesheet linked to in `index.html` contain contain
rules specific to a single component?

**Comment by chalin** [comment]
Adam: btw, I agree with your observations, including the idea of eliminating the `sample.css` file as you have proposed already. I was just suggesting that we (docs team, including myself) first take a broader look to understand why the current approach exists both for this chapter and the other chapters. (As to the lack of mention of `sample.css`, that could be quickly fixed by adding an appropriate sentence to the prose.)

**Comment by Adam-Meisen** [comment]
Sorry if I came off as accusatory, I agree with you.

If it helps, `sample.css` first appeared in [e4db340464f5955670c0ce4e61587875a7d334b3](https://github.com/angular/angular.io/pull/1066/commits/e4db340464f5955670c0ce4e61587875a7d334b3) (merged in PR #1066) and has not been touched by any commit since. It seems that either no one noticed it in the PR, or no one thought it was worth mentioning.

**Comment by chalin** [comment]
No worries (I could have been clearer from the start). Thanks for that info.

**Comment by kwalrath** [comment]
Any objections to me merging this change? @Foxandxss?

**Comment by Foxandxss** [comment]
Works for me.

**Pull request merged**